CMakeLists.txt

```diff
@@ -747,64 +747,81 @@ if(MINGW OR MSYS)
     set(ZLIB_DLL_SRCS ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
 endif()
 
-add_library(zlib SHARED ${ZLIB_SRCS} ${ZLIB_ARCH_SRCS} ${ZLIB_ASMS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
-target_include_directories(zlib PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
-add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_ARCH_SRCS} ${ZLIB_ASMS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
-target_include_directories(zlibstatic PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+if(NOT DEFINED BUILD_SHARED_LIBS)
+    add_library(zlib SHARED  ${ZLIB_SRCS} ${ZLIB_ARCH_SRCS}
+        ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
+    add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_ARCH_SRCS}
+        ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
 
-if(WITH_GZFILEOP)
-    target_sources(zlib PRIVATE ${ZLIB_GZFILE_SRCS})
-    target_sources(zlibstatic PRIVATE ${ZLIB_GZFILE_SRCS})
+    set(ZLIB_INSTALL_LIBRARIES zlib zlibstatic)
+else()
+    add_library(zlib ${ZLIB_SRCS} ${ZLIB_ARCH_SRCS}
+        ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})
```

Comment:
Would it perhaps be smart to combine "${ZLIB_SRCS} ${ZLIB_ARCH_SRCS} ${ZLIB_DLL_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS}" into a single variable ${ZLIB_ALL_LIB_SRCS} or similar?
That would avoid repeating that whole 2-line string 3 times in 8 lines of code. It has no functional difference but would make the code more readable.

Reply:
Done